src/net/tls.cc

```diff
@@ -1830,6 +1841,17 @@ class session : public enable_lw_shared_from_this<session> {
         });
     }
 
+    future<sstring> get_selected_alpn_protocol() {
```

Comment:
Better to return std::optional\<sstring> to make it clear when something wasn't negotiated.